### PR TITLE
[STRAT-69] Route module refactored, tests added

### PR DESCRIPTION
**Pull request by ufshk**
- refactored code file -> now two code files in a subfolder data_retrieval within the folder of routemodel
- added tests that pass -- only test the parsing functions, found it very complex to test getters as they are coded now -> potential update: separate query parameter builder from actual getter? would make it easier to test
- README for routemodel directory now updated

### CODE REVIEW

File: routemodel/README.md [hunk 1]
```diff
@@ -1 +1,54 @@
...
+Connects and receives responses from Bing Maps APIs to form route / navigation and elevations profile.
+
+### Navigation
+
+Retrieves data from Bing Maps API pertaining to navigation of route.
+
+1. Initialization parameters
+   - REQUIRED: List of Major Checkpoints along route - also called "waypoints"
+     - enter list of at least 2 waypoints for request to return a good response with usable data
+       - waypoints can only be coordinates as of now: each index of the list will be a dict of a latitude and longitude, the latitude being the key
```
Comment (clarkevandenhoven) [inline routemodel/README.md:16]: :+1:

File: routemodel/data_retrieval/get_elevations.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        elevation_df = pd.DataFrame(columns=headers)
+
+        # loop through waypoints and viawaypoints
+        # to write coordinates into DataFrame
+        counter = 0
+        for index, waypoint in enumerate(self.waypts):
+            for key in waypoint.keys():
+                elevation_df = elevation_df.append({'Latitude': key,
+                                                    'Longitude': waypoint[key],
+                                                    'Elevation': elevations[counter]}, ignore_index = True)
```
Comment (clarkevandenhoven) [inline routemodel/data_retrieval/get_elevations.py:79]: What happened with the formatting here?

File: routemodel/data_retrieval/get_navigation.py [hunk 3]
```diff
@@ -141,57 +105,15 @@ def parse_route_data(self, response):
...
-                pass
-
-
-        # write dataframe into csv and return true
-        elevation_df.to_csv(ELEVATIONS_FILE)
-        print(f'Elevations written to CSV file.')
-        return True
-
-
 if __name__ == "__main__":
```
Comment (clarkevandenhoven) [inline routemodel/data_retrieval/get_navigation.py:112]: Ideally we remove this main that executes code at the bottom of it, and incorporate its functionality into a test :)

File: routemodel/data_retrieval/get_elevations.py (hunk 2 above)
Comment (ufshk) [reply routemodel/data_retrieval/get_elevations.py:79]: I'm not sure I follow, what is the formatting issue here?

Comment (clarkevandenhoven) [reply routemodel/data_retrieval/get_elevations.py:79]: I think it might just be github but it doesn't show ignore index on a new line

Comment (ufshk) [reply routemodel/data_retrieval/get_elevations.py:79]: Wait, it isn't on a new line, missed this comment.

**Pull request merged**